### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️

- [x] Switch to github-actions.
- [x] Update widgets and tests.
- [x] Add final all widgets.

### CODE REVIEW

File: src/FragmentCache.php [hunk 1]
```diff
@@ -93,11 +99,11 @@ public function start(): void
      */
     public function run(): string
     {
-        if (($content = $this->getCachedContent()) !== false) {
+        if (($content = $this->getCachedContent()) !== null) {
             return $content;
         }
 
-        if ($this->cache instanceof CacheInterface) {
+        if ($this->cache instanceof CacheInterface && $this->noCache === false) {
```
Comment (samdark) [inline src/FragmentCache.php:106]: Both checks are redundant. There's no sense in having no cache and `$this->cache` is always defined and an instance of `CacheInterface`. You can pass a dummy cache here if needed to turn off caching.

File: src/FragmentCache.php [hunk 2]
```diff
@@ -112,36 +118,29 @@ public function run(): string
...
-    public function getCachedContent()
+    public function getCachedContent(): ?string
     {
-        if ($this->content !== null) {
-            return $this->content;
-        }
-
-        if (!($this->cache instanceof CacheInterface)) {
-            return $this->content;
+        if (!($this->cache instanceof CacheInterface) || $this->noCache) {
```
Comment (samdark) [inline src/FragmentCache.php:131]: Same thing here. No need for it.

File: src/FragmentCache.php [hunk 3]
```diff
@@ -160,28 +159,14 @@ public function calculateKey()
...
      *
      * @param string|bool $value
      *
-     * @return FragmentCache
+     * @return $this
      */
-    public function content($value): FragmentCache
+    public function content($value): self
     {
         $this->content = $value;
```
Comment (samdark) [inline src/FragmentCache.php:171]: `$value` is declared as `string|bool` but it seems that's not correct according to property declaration.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/FragmentCache.php [hunk 4]
```diff
@@ -57,10 +61,12 @@ class FragmentCache extends Widget implements DynamicContentAwareInterface
      */
     private $variations;
 
+    private bool $noCache = false;
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FragmentCache.php:64]: if it doesn't make sense to use the widget and not use the cache.

File: src/FragmentCache.php (hunk 1 above)
Comment (terabytesoftw) [reply src/FragmentCache.php:106]: fixed.

File: src/FragmentCache.php (hunk 3 above)
Comment (terabytesoftw) [reply src/FragmentCache.php:171]: fixed.

File: src/FragmentCache.php (hunk 2 above)
Comment (terabytesoftw) [reply src/FragmentCache.php:131]: Fixed.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**